src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProvider.cs

```diff
@@ -75,7 +75,13 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
             if (SyntaxFacts.GetContextualKeywordKind(targetToken.ValueText) == SyntaxKind.AsyncKeyword)
                 return SpecializedTasks.EmptyImmutableArray<SymbolAndSelectionInfo>();
 
-            var typeNode = targetToken.Parent as TypeSyntax;
+            var potentialTypeNode = targetToken.Parent;
+            while (potentialTypeNode is TypeArgumentListSyntax typeArgumentList)
```

Comment:
why do you need the while look?

Reply:
can you ensure that  this only happens when the target token is the `>` token.

Reply:
It could also be an `if` because the target token is always the right-most closing angled bracket, so the immediate parent will always be the GenericNameSyntax, right?

Reply:
that seems sensible to me :)

Reply:
Converted to `if` and added a check for the `>` token.